src/PointwiseFunctions/AnalyticSolutions/Poisson/Moustache.cpp

```diff
@@ -0,0 +1,76 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/AnalyticSolutions/Poisson/Moustache.hpp"
+
+#include <cmath>
+#include <pup.h>  // IWYU pragma: keep
+
+#include "DataStructures/DataVector.hpp"     // IWYU pragma: keep
+#include "DataStructures/Tensor/Tensor.hpp"  // IWYU pragma: keep
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/StdArrayHelpers.hpp"
+// IWYU pragma: no_forward_declare Tensor
+
+namespace Poisson {
+namespace Solutions {
+
+template <size_t Dim>
+tuples::TaggedTuple<Field> Moustache<Dim>::field_variables(
+    const tnsr::I<DataVector, Dim>& x) const noexcept {
+  auto field = make_with_value<Scalar<DataVector>>(x, 1.);
+  for (size_t d = 0; d < Dim; d++) {
+    field.get() *= x.get(d) * (1. - x.get(d));
+  }
+  auto norm_square = make_with_value<DataVector>(get<0>(x), 0.);
+  for (size_t d = 0; d < Dim; d++) {
+    norm_square += square(x.get(d) - 0.5);
+  }
+  field.get() *= pow(norm_square, 3. / 2.);
+  return {std::move(field)};
+}
+
+template <>
+tuples::TaggedTuple<::Tags::Source<Field>> Moustache<1>::source_variables(
+    const tnsr::I<DataVector, 1>& x) const noexcept {
+  auto source = make_with_value<Scalar<DataVector>>(x, 1.);
+  source.get() *= (0.875 - 8.5 * get<0>(x) + 28.5 * square(get<0>(x)) -
```

Comment:
Is there a reason to not define x1 = get<0>(x) here as well? (since you do this in Moustache<2>)